common/lib/api-client/request-headers.js

```diff
@@ -3,13 +3,21 @@ const uuid = require('uuid')
 const {
   API: { VERSION },
   APP_VERSION,
+  CUSTOM_HEADERS,
 } = require('../../../config')
 
-module.exports = (format = 'application/vnd.api+json') => {
-  return {
+module.exports = (req, format = 'application/vnd.api+json') => {
+  const headers = {
     'User-Agent': `hmpps-book-secure-move-frontend/${APP_VERSION}`,
     Accept: `${format}; version=${VERSION}`,
     'Accept-Encoding': 'gzip',
     'Idempotency-Key': uuid.v4(),
+    ...CUSTOM_HEADERS,
   }
+
+  if (req && req.user) {
```

Comment:
So my thinking on dynamic headers is more like what's happening here.

For example if we had `scenario-id` or `k8-ingress-id` or `x-api-debug` on the `req`, and we were configured to allow them through, then we'd add them onto the headers of any downstream requests. But these would be dynamic, and not necessarily present on every request so couldn't entirely be backed into the static config.